src/ios-deploy/ios-deploy.m

```diff
@@ -959,13 +963,27 @@ void write_lldb_prep_cmds(AMDeviceRef device, CFURLRef disk_app_url) {
 void
 server_callback (CFSocketRef s, CFSocketCallBackType callbackType, CFDataRef address, const void *data, void *info)
 {
-    if (CFDataGetLength (data) == 0) {
+    char buffer[0x4000];
+    int bytesRead = AMDServiceConnectionReceive(dbgServiceConnection, buffer, sizeof(buffer));
+    if (bytesRead == 0)
+    {
         // close the socket on which we've got end-of-file, the server_socket.
         CFSocketInvalidate(s);
         CFRelease(s);
         return;
     }
-    write(CFSocketGetNative(lldb_socket), CFDataGetBytePtr(data), CFDataGetLength(data));
+    write(CFSocketGetNative (lldb_socket), buffer, bytesRead);
+    if (bytesRead == sizeof(buffer))
+    {
+        while (bytesRead > 0)
+        {
+            bytesRead = AMDServiceConnectionReceive(dbgServiceConnection, buffer, sizeof(buffer));
```

Comment:
Should also memset before reusing.

Reply:
No point in memsetting as we are always working with ptr + length not c strings. There is no benefit from memeset, only extra work.

Reply:
Oops, you're right. Was assuming we were strlen'ing the buffer when writing back out but bytesRead already gives that information.